app/models/product.rb

```diff
@@ -20,6 +20,10 @@ class Product < ApplicationRecord
   has_many :orders, through: :line_items
   before_destroy :ensure_not_referenced_by_any_line_item
 
+  after_initialize :set_name_default
```

Comment:
dont use after_initialize. See if there's any other callback we can use

Reply:
done using before_save

Reply:
done